packages/zoltan2/core/src/algorithms/zoltan/Zoltan2_AlgZoltan.hpp

```diff
@@ -234,7 +234,41 @@ class AlgZoltan : public Algorithm<Adapter>
     // zz->Set_HG_Edge_Wts_Fn(zoltanHGSizeEdgeWts_withMeshAdapter<Adapter>,
     //                         (void *) &(*adp));
   }
-  
+
+  //! \brief set callbacks for hierarchical coordinate partitioning (e.g. RCB)
+  void setCallbacksHierTaskMapping(
+      const RCP<const MachineRepresentation<scalar_t, part_t> > &machine)
+  {
+    zz->Set_Hier_Num_Levels_Fn(zoltanHierNumLevels<Adapter>, (void *) &(*machine));
+    zz->Set_Hier_Part_Fn(zoltanHierPart<Adapter>, (void *) &(*machine));
+    zz->Set_Hier_Method_Fn(zoltanHierMethod<Adapter>, (void *) &(*machine));
+  }
+
+
+  //! \brief  set callbacks for hierarchical graph partitioning (e.g. ParMETIS)
+  void setCallbacksHierTaskMapping_withGraphModel(
+      const RCP<const GraphAdapter<user_t,userCoord_t> > &adp,
+      const RCP<const MachineRepresentation<scalar_t, part_t> > &machine)
+  {
+    typedef GraphModel<GraphAdapter<user_t, userCoord_t>>  graphmodel_t;
+
+    Zoltan2::modelFlag_t graphFlags;
+
+    graphFlags.set(GENERATE_CONSECUTIVE_IDS);
+    graphFlags.set(REMOVE_SELF_EDGES);
+
+    graphmodel_t *mdl = new graphmodel_t(adp, env, problemComm, graphFlags);
```

Comment:
This declaration will leak memory and be risky when the callbacks accessing the model are actually called.  When this function exits, mdl goes out of scope; there is no way to free it.  But the callbacks are storing pointers to this memory.
The field mdl should probably be a private member of the class.  It should probably be a RCP so that it will be freed when the class is freed.